core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroup.java

```diff
@@ -141,6 +180,9 @@ protected final void addEndpoint(Endpoint e) {
      * Removes the specified {@link Endpoint} from current {@link Endpoint} list.
      */
     protected final void removeEndpoint(Endpoint e) {
+        if (!allowEmptyEndpoints && Iterables.size(endpoints) == 1) {
```

Comment:
super nit; feel free to ignore this comment

I'm wondering why we're using guava's `Iterables` instead of getting size directly